src/lib/scheduler/abstract_task.hpp

```diff
@@ -83,9 +83,8 @@ class AbstractTask : public std::enable_shared_from_this<AbstractTask> {
   void join();
 
   /**
-   * Atomically marks the Task as scheduled, thus making sure this happens only once
+   * @return The Task was scheduled
```

Comment:
This comment sounds weird. Either "return whether the task was scheduled" or just leave it out.